CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -0,0 +1,535 @@
+                    }
+                });
+           
+            dashbordRepositoryMock.Setup(x => x.GetStudentMarksListByStudentIds(studentIdWithoutGroup, new List<long> { }))
+                .ReturnsAsync(new List<StudentMarkDto>());
+
+            _unitOfWorkMock.Setup(x => x.DashboardRepository).Returns(dashbordRepositoryMock.Object);
+
+            var dashbordService = new DashboardService(_unitOfWorkMock.Object);
+
+            //Act
+            var resultWithData = await dashbordService.GetStudentClassbookAsync(studentIdWithGroup, dashbordAnaliticRequstWithData, claimPrincipal); // вопрос о третьем параметре 
+            var resultWithoutClassbook = await dashbordService.GetStudentClassbookAsync(studentIdWithGroup, dashbordAnaliticRequstWithoutClassbook, claimPrincipal);
+            var requestWithStrangerCredentials = await dashbordService.GetStudentClassbookAsync(studentIdWithoutGroup, dashbordAnaliticRequstWithData, claimPrincipal); // вопрос о третьем параметре 
+            var resultWithoutGrop = await dashbordService.GetStudentClassbookAsync(studentIdWithoutGroup, dashbordAnaliticRequstWithData, claimPrincipalForStudentWitoutGroup);
+
+            //Assert
+            Assert.Equal(ErrorCode.ValidationError, requestWithStrangerCredentials.Error.Code);
+            Assert.NotNull(resultWithData);
+            Assert.Equal(ErrorCode.ValidationError, resultWithoutClassbook.Error.Code);
+            Assert.Empty(resultWithoutGrop.Data.StudentsMarks);
+            Assert.Empty(resultWithoutGrop.Data.StudentsPresences);
+        }
+
+        [Fact]
+        public async Task GetStudentResult()
+        {
+            //Arrange
+            long studentIdWithGroup = 7;
+            long studentIdWithoutGroup = 20;
+
+            var claims = new List<Claim>()
+            {
+                new Claim(ClaimTypes.Name, "Denys"),
+                new Claim(ClaimTypes.NameIdentifier , "1"),
+                new Claim(ClaimTypes.Role, UserRole.Student.ToString()),
+                new Claim("Id", "7")
+            };
+
+            var claimIdentity = new ClaimsIdentity(claims, "TestAuthType");
+            var claimPrincipal = new ClaimsPrincipal(claimIdentity);
+
+            var claimsForStudentWitoutGroup = new List<Claim>()
+            {
+                new Claim(ClaimTypes.Name, "Denys"),
+                new Claim(ClaimTypes.NameIdentifier , "1"),
+                new Claim(ClaimTypes.Role, UserRole.Student.ToString()),
+                new Claim("Id", "20")
+            };
+
+            var claimIdentityForStudentWitoutGroup = new ClaimsIdentity(claimsForStudentWitoutGroup, "TestAuthType");
+            var claimPrincipalForStudentWitoutGroup = new ClaimsPrincipal(claimIdentityForStudentWitoutGroup);
+
+            var mockPrincipal = new Mock<IPrincipal>();
+            mockPrincipal.Setup(x => x.Identity).Returns(claimIdentity);
+            mockPrincipal.Setup(x => x.Identity).Returns(claimIdentityForStudentWitoutGroup);
```

Comment:
I am not sure about this is right.... Maybe mockPrincipal has to be created on different cleaims? so you will give it like parameter to new Mock<IPrincipal>()
But line 372 just rewrites what line 371 will do...